homeassistant/components/onewire/onewirehub.py

```diff
@@ -57,116 +55,98 @@ def _is_known_device(device_family: str, device_type: str | None) -> bool:
 class OneWireHub:
     """Hub to communicate with server."""
 
-    def __init__(self, hass: HomeAssistant) -> None:
+    owproxy: protocol._Proxy
+    devices: list[OWDeviceDescription]
+
+    def __init__(self, hass: HomeAssistant, config_entry: OneWireConfigEntry) -> None:
         """Initialize."""
-        self.hass = hass
-        self.owproxy: protocol._Proxy | None = None
-        self.devices: list[OWDeviceDescription] | None = None
-
-    async def connect(self, host: str, port: int) -> None:
-        """Connect to the server."""
-        try:
-            self.owproxy = await self.hass.async_add_executor_job(
-                protocol.proxy, host, port
-            )
-        except protocol.ConnError as exc:
-            raise CannotConnect from exc
+        self._hass = hass
+        self._config_entry = config_entry
 
-    async def initialize(self, config_entry: OneWireConfigEntry) -> None:
-        """Initialize a config entry."""
-        host = config_entry.data[CONF_HOST]
-        port = config_entry.data[CONF_PORT]
+    def _initialize(self, host: str, port: int) -> None:
+        """Connect to the server, and discover connected devices.
+
+        Needs to be run in executor.
+        """
         _LOGGER.debug("Initializing connection to %s:%s", host, port)
-        await self.connect(host, port)
-        await self.discover_devices()
-        if TYPE_CHECKING:
-            assert self.devices
-        # Register discovered devices on Hub
-        device_registry = dr.async_get(self.hass)
+        self.owproxy = protocol.proxy(host, port)
+        self.devices = _discover_devices(self.owproxy)
+
+    async def initialize(self) -> None:
+        """Initialize a config entry."""
+        await self._hass.async_add_executor_job(
+            self._initialize,
+            self._config_entry.data[CONF_HOST],
+            self._config_entry.data[CONF_PORT],
+        )
```

Comment:
Would it make sense to just get the host and port from the entry data directly instead of passing as parameter?